mars/services/scheduling/supervisor/assigner.py

```diff
@@ -61,15 +65,18 @@ async def __pre_destroy__(self):
     async def assign_subtasks(self, subtasks: List[Subtask]):
         inp_keys = set()
         selected_bands = dict()
+        available_bands = list(await self._slots_ref.get_available_bands())
         for subtask in subtasks:
             if subtask.expect_bands:
-                selected_bands[subtask.subtask_id] = subtask.expect_bands
+                selected_bands[subtask.subtask_id] = \
+                    subtask.expect_bands if subtask.expect_bands in available_bands \
```

Comment:
`expect_bands` is a tuple/list and `available_bands` is a list, `subtask.expect_bands in available_bands` will always return False